File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)

        # brute force
        # for i in range(n):
        #     for j in range(i+1,n):
        #         if nums[i]+nums[j] == target:
        #             return [i,j]

        # return []

        # one pass -> hashmap

        mapp = {}

        for idx in range(n):
            num = nums[idx]
            if mapp.get(target - num,-1) > -1:
                return [mapp[target - num],idx]
            mapp[num] = idx
        
        return []

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size(), idx2 = 0;
        map <int,int> mapp;

        for(int i=0;i<n;i++){
            if(mapp[target-nums[i]]>0){
                idx2 = i;
                break;
            }
            mapp[nums[i]] = i;
        }
        
        return {mapp[target - nums[idx2]],idx2};
    }
};